core/src/test/java/com/sequenceiq/cloudbreak/service/environment/credential/OpenSshPublicKeyValidatorTest.java

```diff
@@ -0,0 +1,131 @@
+
+class OpenSshPublicKeyValidatorTest {
+
+    private static final String VALID_PUBLIC_KEY_ED25519 = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIMQltFutaGpkyuDLScqHRZtknBd4c/IJCkVsY7WFS+gK";
+
+    private OpenSshPublicKeyValidator underTest;
+
+    @BeforeEach
+    public void setUp() {
+        underTest = new OpenSshPublicKeyValidator();
+    }
+
+    @Test
+    public void validRsaPublicKeyWillNotFail() {
+        underTest.validate(
+                        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                                + "KR495VFmuOepLYz5I8Dn sequence-eu", false
+        );
+    }
+
+    @Test
+    public void validRsaPublicKeyWillNotFailWhenMissingUser() {
+        underTest.validate(
+                        "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                                + "KR495VFmuOepLYz5I8Dn", false
+        );
+    }
+
+    @Test
+    void validEd25519PublicKeyWillNotFailWhenNotFipsMode() {
+        underTest.validate(VALID_PUBLIC_KEY_ED25519, false);
+    }
+
+    @Test
+    void validEd25519PublicKeyWillFailWhenFipsMode() {
+        BadRequestException badRequestException = assertThrows(BadRequestException.class, () -> underTest.validate(VALID_PUBLIC_KEY_ED25519, true));
+        assertThat(badRequestException).hasMessageEndingWith("detailed message: SSH2ED25519: this key type is not allowed when running clusters in FIPS mode");
+    }
+
+    @Test
+    public void inValidPublicKeyWillFailWhenMissingSshRsa() {
+        BadRequestException badRequestException = assertThrows(BadRequestException.class, () -> underTest.validate(
+                "AAAAB3NzaC1yc2EAAAADAQABAAABAQCasJyap4swb4Hk4xOlnF3OmKVwzmv2e053yrtvcUPaxCeboSltOBReuT"
+                + "QxX+kYCgKCdtEwpIvEDXk16T6nCI4tSptAalFgpUWn+JOysCuLuWnwrk6mSKOzEiPYCrB54444mDY6rbBDSRuE/V"
+                + "UYQ/yi0imocARlOiFdPRlZGTN0XGE1V8LSo+m0oIzTwBKn58I4v5iB4ZUL/6adGXo7dgdBh/Fmm4uYbgrCZnL1EaK"
+                + "pMxSG76XWhuzFpHjLkRndz88ha0rB6davag6nZGdno5IepLAWg9oB4jTApHwhN2j1rWLN2y1c+pTxsF6LxBiN5rsY"
+                + "KR495VFmuOepLYz5I8Dn sequence-eu", false
```

Comment:
Hint: Might use """ textblock quote

Reply:
It is a single line without linebreaks. :smile: